### PR TITLE
Implement 3 RPCs for metrics, register metric service

### PR DESCRIPTION
**Pull request by the-ericwang35**

### CODE REVIEW

File: server/metric/mock_metric.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/genproto/googleapis/rpc/status"
+)
+
+type MockMetricServer struct {
+	monitoring.UnimplementedMetricServiceServer
+}
+
+func (s *MockMetricServer) ListMonitoredResourceDescriptors(
```
Comment (mabdi3) [inline server/metric/mock_metric.go]: I think it's fine to have the method header be on one line

**Review by mabdi3 — CHANGES_REQUESTED** [review]
👍 from me, small nit.

File: server/metric/mock_metric.go (hunk 1 above)
Comment (the-ericwang35) [reply server/metric/mock_metric.go]: Sure, changed

**Review by mabdi3 — APPROVED** [review]
LGTM

**Pull request merged**